### ISSUE TITLE
`nf-core bump-version` breaks linting

### ISSUE THREAD

**Issue by d4straub** [nf-core/tools#1265]
## Description of the bug

`nf-core bump-version` removes the last newline in `nextflow.config` and therefore breaks linting, I had to correct that for [ampliseq](https://github.com/d4straub/ampliseq/commit/84d228850303dc565a92ef204ff9f2b82c6e0a44) and [bacass](https://github.com/nf-core/bacass/commit/409954c13a45e768f9e18de3be4bf1dd5097f895) after bumping the version.

## Steps to reproduce

Steps to reproduce the behaviour:

nf-core bump-version <anything>
results: nf-core linting on github `EditorConfig` will fail with 

```console
$ editorconfig-checker -exclude README.md $(git ls-files | grep -v test)
nextflow.config:
	Wrong line endings or new final newline
1 errors found
```

## Version

nf-core/tools version 2.1

**Comment by ewels** [comment]
I don't really understand how this can happen 🤔 We do a regex find + replace for a specific string (that is not at the end of the file). Is Python really messing with the newlines just from opening and resaving a file?

**Comment by ewels** [comment]
Tested and replicated the error. It's my fault, it's because the code splits the file into an array of lines and then joins them with a newline. This means that the trailing newline is removed.


Fixed in #1438
